### PR TITLE
drivers: pwm: stm32: remove remaining Kconfig instances

### PR DESCRIPTION
**Pull request by gmarull**
Following other drivers, Kconfig based instances are now entirely removed. In order to do this change, PWM nodes in board DT files have been given a pwm{N} label so that both:

- DT API checks such as `#if DT_HAS_NODE(DT_NODELABEL(pwmN))` can be  used (`N` being PWM instance number).
- DT references can be written as `pwms = <&pwmN x y>;` instead of `pwms = <&{/soc/timers@XXXXXXXX/pwm} x y>;`

This approach is also used on the Linux Kernel. Reference: https://wiki.st.com/stm32mpu/wiki/TIM_device_tree_configuration#DT_configuration_examples

### CODE REVIEW

File: boards/arm/stm32f746g_disco/pinmux.c [hunk 1]
```diff
@@ -22,9 +22,9 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PC6, STM32F7_PINMUX_FUNC_PC6_USART6_TX},
 	{STM32_PIN_PC7, STM32F7_PINMUX_FUNC_PC7_USART6_RX},
 #endif
-#ifdef CONFIG_PWM_STM32_3
+#if DT_HAS_NODE(DT_NODELABEL(pwm2))
```
Comment (erwango) [inline boards/arm/stm32f746g_disco/pinmux.c:25]: pwm3

Comment (erwango) [reply boards/arm/stm32f746g_disco/pinmux.c:25]: This being said, in device tree only pwm1 is enabled...

**Review by erwango — CHANGES_REQUESTED** [review]
One comment, otherwise I'm fine with this proposal.
Also, it good to detail all the approaches in PR description, but maybe having them in commit message could bring confusion later on.

File: boards/arm/stm32f746g_disco/pinmux.c (hunk 1 above)
Comment (gmarull) [reply boards/arm/stm32f746g_disco/pinmux.c]: Fixed. Looks like pwm3 should be the enabled node. I can add a separate commit to fix that, even though it is out of the scope of this PR.

**Comment by gmarull** [comment]
@erwango I have updated the PR with an approach that does not require SoC DTS changes and actually is the approach given by ST on Linux Kernel.

**Comment by gmarull** [comment]
@henrikbrixandersen I have finally updated PR to follow the ST approach on Linux Kernel (see link in the description). I think it's the solution that for now has the smallest impact and at the same time follows the same convention as on Linux Kernel.

**Comment by gmarull** [comment]
@erwango 4ce5ee1 is just a hack suggested by @carlescufi  to pass CI (it is timing out due to the large number of tests). It should be removed before merging.

**Review by erwango — APPROVED** [review]
Thanks for this change!

**Comment by gmarull** [comment]
@erwango is it ok to remove CI fix now?

**Comment by erwango** [comment]
> @erwango look at the STM32 board renames, I fixed a few, but many have the wrong string for the name in the board yaml file.

@galak will tell you

**Comment by galak** [comment]
passed CI, removed the CI workaround.  Going to merge by hand.

**Pull request merged**